pkg/homedir/homedir.go

```diff
@@ -34,6 +38,31 @@ func GetDataHome() (string, error) {
 		return "", errors.New("could not get either XDG_DATA_HOME or HOME")
 	}
 	return filepath.Join(home, ".local", "share"), nil
+
+}
+
+// GetDataHomeByUID finds the home directory for the given uid, only works with sudo permissions
+func GetDataHomeByUID(uid int) (string, error) {
+	var home string
+	u, err := user.LookupId(fmt.Sprint(uid))
+	if err != nil {
+		return "", err
+	}
+	cmd := exec.Command("sudo", "-Hiu", u.Username, "env")
+	out, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	env := strings.Split(string(out), "\n")
+	for _, val := range env {
+		if strings.Contains(val, "XDG_DATA_HOME") || strings.Contains(val, "HOME") {
```

Comment:
for what functionality do we need this function?

Reply:
I think it is risky to add such a function to c/storage as we are stuck with it forever.

This function cannot be used from a user namespace since `sudo` doesn't work there.

Maybe we could add it to Podman for now (or ignore `XDG_DATA_HOME` to detect rootless storage directories)?

I suggest to ignore `XDG_DATA HOME` since anyway it is a heuristic as the user can override it at any time, and I don't think we can detect it.  If the store is not present under `$HOME/.local/share/containers/storage` then handle it as not detected

Reply:
I am making a `podman system setup` feature that has the ability to take a user and setup podman storage from scratch. When run as root, `types.DefaultStorOptions` returns the root store options even if `rootless = true` is passed.

Reply:
the issue here is I have no way of getting the user's home directory from podman if we are running as root trying to find another user's default storage options.

Reply:
Why not use machinectl login ... And execute podman in the user environment, falling back to sudo? I think this is the safest way forward. One issue with doing this via Code is you might miss things like storage.conf and containers.conf files sitting in the users homedir. Seems to me is what you want is to execute Podman in the same way a user would.

Reply:
@rhatdan would it make more sense to use `machinectl` within podman to avoid issues @giuseppe brings up about long-term maintenance w/in storage? I could basically create this function with machinectl and sudo in podman to get the home directory, and then modify the `DefaultStoreOption`'s run/graph root accordingly once they are returned?

Reply:
I believe this function should be moved to Podman, and we should default to machinectl and fall back to sudo.